internal/backend/remote/backend.go

```diff
@@ -582,7 +582,7 @@ func (b *Remote) WorkspaceNamePattern() string {
 }
 
 // DeleteWorkspace implements backend.Enhanced.
-func (b *Remote) DeleteWorkspace(name string) error {
+func (b *Remote) DeleteWorkspace(name string, _ bool) error {
```

Comment:
I am not 100% sure if it makes sense to also add usage of the Safe Delete API to the `remote` backend (instead of just the `cloud` backend). I have not in this PR so far, but input is appreciated

Reply:
I don't think it's strictly necessary to backport your changes from cloud to backend "remote" but it depends on how many terraform users you'd like to reach with this feature.

Reply:
My instinct then is to leave `remote` as it is, since:
- we recommend `cloud` instead
- the CLI already does some safe vs force delete checks, this change is just some additional guards. So having the `remote` backend continue to fallback to the existing force-delete APIs seems 👌